### PR TITLE
World generation mappings for 21w13a

### PR DESCRIPTION
**Pull request by mschae23**
Some mappings for the world generation changes in 21w13a.

### CODE REVIEW

**Comment by liach** [comment]
Conflicts with #2221

**Review by jaskarth — APPROVED** [review]

**Comment by liach** [comment]
@mschae23 Mind resolve the merge conflicts?

**Comment by mschae23** [comment]
I resolved the merge conflict. Hopefully I did that correctly.

**Review by liach — APPROVED** [review]
imo valueprovider package can be renamed gen, as they generate numbers

File: mappings/net/minecraft/util/math/valueprovider/IntProviderType.mapping [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CLASS net/minecraft/class_6018 net/minecraft/util/math/valueprovider/IntProviderType
```
Comment (YanisBft) [inline mappings/net/minecraft/util/math/valueprovider/IntProviderType.mapping:1]: Same for this one

**Review by enbrain — COMMENTED** [review]
Can you resolve conflicts between `net/minecraft/util/math/intprovider` and `net/minecraft/util/math/valueprovider`? This is something Git can't detect.

File: mappings/net/minecraft/util/math/valueprovider/IntProvider.mapping [hunk 2]
```diff
@@ -0,0 +1,13 @@
+CLASS net/minecraft/class_6017 net/minecraft/util/math/valueprovider/IntProvider
```
Comment (YanisBft) [inline mappings/net/minecraft/util/math/valueprovider/IntProvider.mapping:1]: Same for this one

File: mappings/net/minecraft/util/math/valueprovider/UniformIntProvider.mapping [hunk 3]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/class_6019 net/minecraft/util/math/valueprovider/UniformIntProvider
```
Comment (YanisBft) [inline mappings/net/minecraft/util/math/valueprovider/UniformIntProvider.mapping:1]: Same for this one

File: mappings/net/minecraft/util/math/valueprovider/ConstantIntProvider.mapping [hunk 4]
```diff
@@ -0,0 +1,11 @@
+CLASS net/minecraft/class_6016 net/minecraft/util/math/valueprovider/ConstantIntProvider
```
Comment (YanisBft) [inline mappings/net/minecraft/util/math/valueprovider/ConstantIntProvider.mapping:1]: This class was already mapped, in a different package tho

File: mappings/net/minecraft/util/math/weight/WeightedList.mapping [hunk 5]
```diff
@@ -0,0 +1,16 @@
+CLASS net/minecraft/class_6012 net/minecraft/util/math/weight/WeightedList
```
Comment (YanisBft) [inline mappings/net/minecraft/util/math/weight/WeightedList.mapping:1]: There is already a class named WeightedList in nm/util/collection. What's the difference(s)?

**Review by YanisBft — CHANGES_REQUESTED** [review]

**Comment by liach** [comment]
Height provider and CarverConfig superseded by #2292; weighting covered by #2281, so sorry!

**Pull request closed without merging**